kotlinx-coroutines-core/common/src/channels/Channel.kt

```diff
@@ -117,21 +126,21 @@ public interface SendChannel<in E> {
      * }
      *
      * val uiUpdater = launch(Dispatchers.Main, parent = UILifecycle) {
-     *    events.consume {}
-     *    events.cancel()
+     *    events.consume { ... handle events ... }
      * }
-     *
+     * // Stop callback as soon as the channel is closed or cancelled
      * events.invokeOnClose { callbackBasedApi.stop() }
      * ```
      *
-     * **Note: This is an experimental api.** This function may change its semantics, parameters or return type in the future.
+     * **Stability note.** This function constitutes a stable API surface with the only exception
+     * about [IllegalStateException] being thrown when multiple handlers are registered.
+     * There is a possibility that this restriction will be lifted in the future.
```

Comment:
Why so formal? I thought we were going for a bit of a hip and easy-to-read feeling with our documentation.
```suggestion
     * This restriction could be lifted in the future.
```